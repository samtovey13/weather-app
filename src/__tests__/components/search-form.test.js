import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import SearchForm from '../../components/search-form';

describe('SearchForm', () => {
const handleSearchText = jest.fn();
const errorMessage = "";

  it('renders correctly', () => {
    const { asFragment } = render(<SearchForm handleSearchText={handleSearchText} errorMessage={errorMessage}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls the onClick function with text input from the user', () => {
    const { getByRole } = render(<SearchForm handleSearchText={handleSearchText} errorMessage={errorMessage}/>);
    const searchInput = getByRole('searchbox');
    const searchButton = getByRole('button');

    fireEvent.change(searchInput, {
      target: { value: "another city" }
    })
    fireEvent.click(searchButton);

    expect(handleSearchText).toHaveBeenCalledTimes(1);
    expect(handleSearchText).toHaveBeenCalledWith("another city");
  });

  describe('renders the error message only when required', () => {
    it('renders the error message when it exists', () => {
      const { getByText, getByTestId } = render(<SearchForm handleSearchText={handleSearchText} errorMessage={"Oops! There is an error."}/>);
      expect(getByText("Oops! There is an error.")).toBeInTheDocument();
      expect(getByTestId("error-message")).toBeInTheDocument();
    });

    it('hides the error message when it is empty', () => {
      const { queryByTestId } = render(<SearchForm handleSearchText={handleSearchText} errorMessage={""}/>);
      expect(queryByTestId("error-message")).not.toBeInTheDocument();
    })
  })
})