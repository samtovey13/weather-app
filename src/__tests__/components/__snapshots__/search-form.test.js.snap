// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <div
      id="search-input-wrapper"
    >
      <input
        class="search-input"
        id="location-search-input"
        placeholder="Find a city"
        type="search"
      />
      <button
        class="search-button"
        data-testid="search-button"
      >
        <i
          class="fa fa-search"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
